elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCancelThread.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.TransactionRegistry;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Runnable thread for cancelling AsyncQuery transactions
+ * beyond the max run time or if it has status CANCELLED.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class AsyncQueryCancelThread implements Runnable {
+
+    private int maxRunTimeSeconds;
+    private Elide elide;
+    private AsyncQueryDAO asyncQueryDao;
+
+    @Override
+    public void run() {
+        cancelAsyncQuery();
+    }
+
+    /**
+     * This method cancels queries based on threshold.
+     */
+    protected void cancelAsyncQuery() {
+
+        try {
+            TransactionRegistry transactionRegistry = elide.getTransactionRegistry();
+            Map<UUID, DataStoreTransaction> runningTransactionMap = transactionRegistry.getTransactionMap();
+            Collection<AsyncQuery> asyncQueryCollection = asyncQueryDao.getActiveAsyncQueryCollection();
+
+            for (Map.Entry<UUID, DataStoreTransaction> entry : runningTransactionMap.entrySet()) {
```

Comment:
This is a bit of pseudocode, but I think the code is more readable with functional style and breaking this into sets:

```java

Set<UUID> runningQueries = runningTransctionMap.keySet();

Set<UUID> cancellingQueries = asyncQueryCollection.stream()
     .filter(query -> query.getStatus == QueryStatus.CANCELED)
     .map (AsyncQuery::getRequestId)
      .collect(Collectors.toSet());

Set<UUID> queriesToKill = Sets.intersection(runningQueries, cancellingQueries);

queriesToKill.stream().
   .map(transactionRegistry.getRunningTransaction)
   .forEach((tx) -> {
      tx.cancel();
   });

```